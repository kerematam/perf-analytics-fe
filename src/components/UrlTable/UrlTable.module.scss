.styled_row {
  &:nth-of-type(odd) {
    background: rgba(0, 0, 0, 0.04);
  }
}

.styled_cell {
  background-color: black;
  color: white;
}

.button {
  color: transparent;
}

.tableOperationContainer {
  $tableItemMargin: 10px;
  margin: 0 -$tableItemMargin;
  min-width: 250px;
  .operationItem {
    margin: 0 $tableItemMargin;
  }
}

.url_cell {
  max-width: 200px;
  overflow: hidden;
  text-overflow: ellipsis;
}
