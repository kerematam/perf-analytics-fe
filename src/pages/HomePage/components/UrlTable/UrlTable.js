import React from 'react'
import {
  Button,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
} from '@material-ui/core'
import CodeIcon from '@material-ui/icons/Code'
import EqualizerIcon from '@material-ui/icons/Equalizer'
import history from 'utils/history'

import styles from './UrlTable.module.scss'

const StyledTableCell = ({ children, ...rest }) => {
  return (
    <TableCell {...rest} className={styles.styled_cell}>
      {children}
    </TableCell>
  )
}

const formatDate = dateStr => new Date(dateStr).toLocaleString()

export default function UrlTable({ pages = [], loading, getPage = () => {} }) {
  return (
    <TableContainer component={Paper}>
      <Table aria-label="customized table">
        <TableHead>
          <TableRow>
            <StyledTableCell>Page URL</StyledTableCell>
            <StyledTableCell>ID</StyledTableCell>
            <StyledTableCell>Create Date</StyledTableCell>
            <StyledTableCell>Operations</StyledTableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {pages.map(row => (
            <TableRow key={row._id} className={styles.styled_row}>
              <TableCell
                title={row.url}
                className={styles.url_cell}
                component="th"
                scope="row"
              >
                {row.url}
              </TableCell>
              <TableCell>{row._id}</TableCell>
              <TableCell>{formatDate(row.createdAt)}</TableCell>
              <TableCell>
                <div className={styles.tableOperationContainer}>
                  <Button
                    onClick={() => {
                      history.push(`/stats/${row._id}`)
                    }}
                    className={styles.operationItem}
                    variant="contained"
                    size="small"
                    startIcon={<EqualizerIcon />}
                  >
                    Stats
                  </Button>
                  <Button
                    onClick={() => {
                      getPage(row)
                    }}
                    className={styles.operationItem}
                    variant="contained"
                    size="small"
                    startIcon={<CodeIcon />}
                  >
                    Script
                  </Button>
                </div>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  )
}
