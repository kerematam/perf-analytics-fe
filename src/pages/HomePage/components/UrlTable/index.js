import UrlTable from './UrlTable'

export default UrlTable
