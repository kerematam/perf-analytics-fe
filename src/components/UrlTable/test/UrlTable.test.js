import React from 'react'
import { render, screen, fireEvent } from '@testing-library/react'
import renderer from 'react-test-renderer'
import UrlTable from '../index'

const MOCK_PROPS = {
  pages: [
    {
      _id: '123123',
      url: 'test',
      createdAt: new Date('11/1/2020'),
    },
  ],
  loading: false,
  getPage: () => {},
}

const setup = setupProps => {
  const wrapper = renderer.create(<UrlTable {...setupProps} />).toJSON()

  return {
    wrapper,
  }
}

describe('UrlTable ', () => {
  test('should match with snapshot', () => {
    const { wrapper } = setup(MOCK_PROPS)
    expect(wrapper).toMatchSnapshot()
  })
})
