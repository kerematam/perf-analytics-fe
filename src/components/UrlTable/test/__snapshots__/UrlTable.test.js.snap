// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UrlTable  should match with snapshot 1`] = `
<div
  className="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
>
  <table
    aria-label="customized table"
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head styled_cell"
          scope="col"
        >
          Page URL
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head styled_cell"
          scope="col"
        >
          ID
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head styled_cell"
          scope="col"
        >
          Create Date
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head styled_cell"
          scope="col"
        >
          Operations
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    >
      <tr
        className="MuiTableRow-root styled_row"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body url_cell"
          scope="row"
          title="test"
        >
          test
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          123123
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          11/1/2020, 12:00:00 AM
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            className="tableOperationContainer"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained operationItem MuiButton-containedSizeSmall MuiButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeSmall"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20h4V4h-4v16zm-6 0h4v-8H4v8zM16 9v11h4V9h-4z"
                    />
                  </svg>
                </span>
                Stats
              </span>
            </button>
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-contained operationItem MuiButton-containedSizeSmall MuiButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeSmall"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
                    />
                  </svg>
                </span>
                Script
              </span>
            </button>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
